### PR TITLE
maintenance: add demo site deployment instructions

### PR DESCRIPTION
**Pull request by ppanero**
@lnielsen @zzacharo 

This is an improved version of what Zach and I used last time to deploy. If needs clarification please let me know :)

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -57,6 +57,8 @@ nav:
     - Configuration: reference/configuration.md
     - REST API: reference/rest_api.md
     - Metadata: reference/metadata.md
+  - Maintenance:
```
Comment (lnielsen) [inline mkdocs.yml:60]: Minor: Should we call it "Developers guide" or "Maintainers guide" instead?

Comment (lnielsen) [reply mkdocs.yml:60]: Btw, let's just discuss on Friday's morning standup.

Comment (lnielsen) [reply mkdocs.yml:60]: Decided in standup: "Maintainers guide"

**Review by lnielsen — APPROVED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (ppanero) [reply mkdocs.yml]: changed :)

**Pull request merged**